packages/eslint-plugin/configs/react.js

```diff
@@ -26,6 +26,5 @@ module.exports = {
 		'react/prop-types': 'off',
 		'react/react-in-jsx-scope': 'off',
 		'react-hooks/rules-of-hooks': 'error',
-		'react-hooks/exhaustive-deps': 'warn',
```

Comment:
Can you explain why this was removed? Looking back through the history to https://github.com/WordPress/gutenberg/pull/14995#discussion_r275943117, I see there was some initial uncertainty about the rule. But as I've been reading more about it, it seems like one of those things where it should be a safe default to use, and that most problems end up from mis-use ([related discussion](https://github.com/facebook/react/issues/14920)). What are these valid exceptions we have, and are they really the majority of cases that we shouldn't only be opting-out as opposed to disabling this rule altogether?

Reply:
When I first started trying hooks, it felt that we had a too many exceptions. Maybe related to the way our components were written or related to my thought process when writing components. happy to reconsider.

Reply:
Truthfully, I'm still trying to understand all that the rule is meant to cover (its documentation... doesn't exist), but the general tone of the discussion around it from their end seems to be that if a developer is fighting with the rule, there's a higher likelihood the developer is "doing it wrong", so I'm curious to see a few specifics.

Maybe I'll flip the rule on locally to see what issues crop up, in order to get a better understanding.

Reply:
It:

- Forces you to pass `deps = []`.
- Statically analyzes callbacks for closures and forces you to add them to `deps`.